### PR TITLE
Quamtum Metric: sendEvent on subscription conversion

### PR DESCRIPTION
**Pull request by GHaberis**
## What are you doing in this PR?

This PR implements a `sendEvent` call for Subscription checkout Conversions. To do this I've created a new export `sendEventSubscriptionCheckoutConversion` from `quantumMetric.ts`, there was a lot of repeated logic between this and `sendEventSubscriptionCheckoutStart` so I pulled out the common code between these two functions into a new  function`sendEventSubscriptionCheckoutEvent` that is internal to `quantumMetric.ts`.

[**Trello Card**](https://trello.com/c/juhvvBXC/491-quantum-metric-implement-sendevent-call-for-subscription-conversion)

### CODE REVIEW

File: support-frontend/assets/helpers/subscriptionsForms/submit.ts [hunk 1]
```diff
@@ -148,32 +149,18 @@ function getGiftRecipient(giftingState: GiftingState) {
...
-		debugInfo,
-	} = state.page.checkout;
-	const addresses = getAddresses(state);
-	const price = getProductPrice(
-		productPrices,
-		addresses.billingAddress.country,
-		billingPeriod,
-		fulfilmentOption,
-		productOption,
-	);
```
Comment (GHaberis) [inline support-frontend/assets/helpers/subscriptionsForms/submit.ts:172]: These `const` have been edited as we're now passing in some of this info as parameters (`addresses` and `promotions`)

**Comment by GHaberis** [comment]
[Quantum Metric: Implement SendEvent call for Subscription Conversion](https://trello.com/c/juhvvBXC/491-quantum-metric-implement-sendevent-call-for-subscription-conversion)

File: support-frontend/assets/helpers/tracking/quantumMetric.ts [hunk 2]
```diff
@@ -5,8 +5,9 @@ import { isSwitchOn } from 'helpers/globalsAndSwitches/globals';
 import type { IsoCurrency } from 'helpers/internationalisation/currency';
 import type { BillingPeriod } from 'helpers/productPrice/billingPeriods';
 import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import type { SubscriptionProduct } from 'helpers/productPrice/subscriptions';
```
Comment (TomPretty) [inline support-frontend/assets/helpers/tracking/quantumMetric.ts:8]: This file is getting a little bigger, could potentially be worth adding some comment headers in to help break up the file slightly? It seems we have: `types`, `sending events`, and `initialisation`? Also, potentially within those sections I personally prefer the most important functions (i.e the ones being exported) right at the top, and then all the various helpers and shared functions that are just internal to the file after

File: support-frontend/assets/helpers/tracking/quantumMetric.ts [hunk 3]
```diff
@@ -104,6 +101,79 @@ function sendEventSubscriptionCheckoutStart(
 	});
 }
 
+function getSubsProductSendEventKey(
```
[the remark this answers is not in the document]
Comment (GHaberis) [reply support-frontend/assets/helpers/tracking/quantumMetric.ts:104]: OK I'll test how this works, both suggestions would remove the existing code for getting the key, which is admittedly a little hard to follow! My preference would be for a single `productToCheckoutEvents` option that return `{ start, conversion }` keys for each product...

File: support-frontend/assets/helpers/tracking/quantumMetric.ts (hunk 2 above)
Comment (GHaberis) [reply support-frontend/assets/helpers/tracking/quantumMetric.ts:8]: I've put some comment headers in for now

Comment (TomPretty) [reply support-frontend/assets/helpers/tracking/quantumMetric.ts:6]: nice - it's looking great!

**Review by TomPretty — APPROVED** [review]
Looks great to me! 🚀
